SDK/AppCenterCrashes/Microsoft.AppCenter.Crashes.Windows.Shared/Crashes.cs

```diff
@@ -101,5 +83,57 @@ private static void PlatformNotifyUserConfirmation(UserConfirmation confirmation
         private static void PlatformTrackError(System.Exception exception, IDictionary<string, string> properties)
         {
         }
+
+        protected override string ChannelName => "crashes";
+
+        public override string ServiceName => "Crashes";
+
+        /// <summary>
+        /// Method that is called to signal start of Crashes service.
+        /// </summary>
+        /// <param name="channelGroup">Channel group</param>
+        /// <param name="appSecret">App secret</param>
+        public override void OnChannelGroupReady(IChannelGroup channelGroup, string appSecret)
+        {
+            lock (CrashesLock)
+            {
+                base.OnChannelGroupReady(channelGroup, appSecret);
+                ApplyEnabledState(InstanceEnabled);
+            }
+        }
+        
+        private void ApplyEnabledState(bool enabled)
+        {
+            lock (_serviceLock)
+            {
+                if (enabled && ChannelGroup != null)
+                {
+                    ApplicationLifecycleHelper.Instance.UnhandledExceptionOccurred += OnUnhandledExceptionOccurred;
+                }
+                else if (!enabled)
+                {
+                    ApplicationLifecycleHelper.Instance.UnhandledExceptionOccurred -= OnUnhandledExceptionOccurred;
+                    ErrorLogHelper.RemoveAllStoredErrorLogFiles();
+                }
+            }
+        }
+
+        public override bool InstanceEnabled
```

Comment:
Just curiosity, what's the coding style/rule for an order of methods/properties?

Reply:
There's no set pattern between modules, but we could move it up below the constructor if you think that would be better

Reply:
It is okay. I'm just curious about any rules that I need to follow.

Reply:
We used `#region` in some files but no clear rules. I guess you can improve and even suggest a new rule.